src/deck.js

```diff
@@ -40,6 +40,41 @@ function points_nine(player) { return 9; }
 function space_zero(player, to_be_added) { return 0; }
 function space_one(player, to_be_added) { return 1; }
 
+const OCCUPANCY = {
```

Comment:
interesting approach, but I am not sure if this turns out easier for the special cards. Or will get special cards their own functions? I am curious.

Reply:
I'm thinking about creating a COMPLEX occupancy, that will take as a parameters list of simple occupancies and combine them together.
For example farm could be written as complex(standard_space, shares_with_greenhouse, shares_with_pigs_power) and summer farm could be written as complex(farm, no_space)

But this requires some more tweaking, first I wanted to get your approval on the idea

Reply:
Let me think about it. I am not sure how this will work out in the end. If then again a greenhouse shares again a space with a farm, which of the functions will occupy the space? I don't yet see, how this should make things easier.

Reply:
Well, right now both Greenhouse and Farm have special functions that check if there is more of A than B ( for farms even more complex, because there is a special farm and there is a dependency on Pigs), then some magic happens to decide if farms or greenhouses should take a space.

In new approach, Farms would always take a space, unless there are pigs, Special Farms would never take a space, Legendary Farm would always take a space and create a space.... And N greenhouses would ALWAYS take exactly (N - ammount(farm)) spaces, or 0

Same for husband/wife combo. Wifes would be a variable here, because there are many implementations of a wife, and husbands would be dependant on ammount of wives.

I hope my daughter will feel better tomorrow, and we're having a long weekend coming up, I can prepare those cases for you

Reply:
Yes, this sounds nice, but when you have these unidirectional dependencies of a card e.g. greenhouse -> farm, wive(s) -> husband etc. you will get a problem once such a _dependency target_ shall be added to a full town, if there would be a "free" _dependency source_.
Maybe the easiest way to check if a card can be added, is to simply:
* add the card to town
* check occupancy against max (even new max with legendary cards)
* **if** violation revert addition and exit
* **else** continue
instead of having this whole _will be added_ information. That maybe would ease things up a bit...
Just some of my ideas.

Nevertheless I wish your daughter a fast recovery 🤞

Reply:
Actually I had a thought or two if the app isn't too strict to be honest. Maybe instead of blocking actions it would be better to show a warning that someone is forgetting about a rule or two? If someone added 16th card somewhere during the gameplay because he forgot about something the app gets a little bit useless.

Maybe let's create a separate issue as a placeholder for future discussion about this? For now I will try to finish my idea ASAP and we will see how that works. But the idea about adding and removing crossed my mind too

Reply:
One more thing that came to my mind....

Right now we have this strange issue, that is not possible to fix I am afraid
1. Fill your city with anything that is not a farm, a greenhouse
2. Try to add a Greenhouse - fails, because of the city limit
3. Add Summer Farm - succeeds, because does not take a space
4. Add a Greenhouse - succeeds, because shares a space with a Summer Farm, thus does not take a space 🤯

Reply:
Yes, this probably happend due to our latest changes with the basecard search. Previously the greenhous did not match with the summerfarm, which makes sense, as she does not need a space, but the greenhouse does.
I feared that your function is pretty dangerous to use...

Reply:
> Actually I had a thought or two if the app isn't too strict to be honest. Maybe instead of blocking actions it would be better to show a warning that someone is forgetting about a rule or two? If someone added 16th card somewhere during the gameplay because he forgot about something the app gets a little bit useless.
> 
> Maybe let's create a separate issue as a placeholder for future discussion about this? For now I will try to finish my idea ASAP and we will see how that works. But the idea about adding and removing crossed my mind too

Well I think your refactor is greatly effected by this decision and maybe should be done together...

Reply:
> Well I think your refactor is greatly effected by this decision and maybe should be done together...

They could, but I thing I am able to make that working, with a couple of TODOs to cleanup later on.

Reply:
> Previously the greenhous did not match with the summerfarm, which makes sense, as she does not need a space, but the greenhouse does.

Actually, that is intended, if greenhouse shares a space with something that does not take a space then it will not take a space either. I saw it in the 'oficial' FAQ and BGG forums answer of the designer:
https://boardgamegeek.com/thread/2953745/greenhouse-and-new-farm
![image](https://github.com/tadeaustria/EverdellScore/assets/471436/23418e1d-6539-42ba-805b-998aa02b1263)

Reply:
Additionally, same issue is possible to replicate with Pigs, standard Farms and Greenhouse for example

Reply:
I don't really see the problem then:
-     Fill your city with anything that is not a farm, a greenhouse **(15)**
-     Try to add a Greenhouse - fails, because of the city limit (would be **16**)
-     Add Summer Farm - succeeds, because does not take a space (**14**, because gh _shares_ no space with summer farm)
-     Add a Greenhouse - succeeds, because shares a space with a Summer Farm, thus does not take a space **(15)**
Although I would not agree on the _sharing no space_ but if the designer says so...

Reply:
Actually this does make sense, as you build the greenhouse on the farm that does not take a space (because it's outside the towns borders for example), then that greenhouse also would be outside the borders. At least I'm trying to understand it this way